relay-server/src/services/processor/replay.rs

```diff
@@ -151,3 +154,168 @@ fn process_replay_event(
 
     Ok(replay)
 }
+
+fn process_replays_combine_items(state: &mut ProcessEnvelopeState) -> Result<(), ProcessingError> {
+    let project_state = &state.project_state;
+    let combined_envelope_items =
+        project_state.has_feature(Feature::SessionReplayCombinedEnvelopeItems);
+
+    if combined_envelope_items {
+        // If this flag is enabled, combine both items into a single item,
+        // and remove the original items.
+        // The combined Item's payload is a MsgPack map with the keys
+        // "replay_event" and "replay_recording".
+        // The values are the original payloads of the items.
+        let envelope = &mut state.envelope_mut();
+        if let Some(replay_event_item) =
+            envelope.take_item_by(|item| item.ty() == &ItemType::ReplayEvent)
+        {
+            if let Some(replay_recording_item) =
+                envelope.take_item_by(|item| item.ty() == &ItemType::ReplayRecording)
+            {
+                let mut data = Vec::new();
+                let mut combined_item_payload = BTreeMap::new();
+
+                combined_item_payload.insert("replay_event", replay_event_item.payload().to_vec());
+                combined_item_payload
+                    .insert("replay_recording", replay_recording_item.payload().to_vec());
+
+                if let Err(e) = rmp_serde::encode::write(&mut data, &combined_item_payload) {
+                    relay_log::error!(
+                        "failed to serialize combined replay event and recording: {}",
+                        e
+                    );
+                    // TODO: what to do here? Drop + emit outcome?
```

Comment:
If we're adding back the original items then I would say no.

Otherwise you can probably use https://github.com/getsentry/relay/blob/c3ac7c0eba8c48587c7fab0621c741f5dc111aee/relay-server/src/utils/managed_envelope.rs#L265.